src/components/DisplayWeb3.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from 'react'
+import ReactDOM from 'react-dom'
+
+import { Web3Api } from '../types'
+
+interface DisplayWeb3Props {
+    web3Api: Web3Api
+}
+
+interface ClientVersion {
+    id?: number
+    jsonrpc?: string
+    result: string
+}
+
+export const DisplayWeb3: React.FC<DisplayWeb3Props> = ({ web3Api }) => {
+    const [account, setAccount] = useState<string>()
+
+    const [balance, setBalance] = useState<string>()
+
+    const [networkName, setNetworkName] = useState<string>()
+
+    const [clientVersion, setClientVersion] = useState<ClientVersion>()
+
+    useEffect(() => {
+        const setter = async (): Promise<void> => {
+            const accountsProm = web3Api.getAccounts()
+            const networkNameProm = web3Api.getNetworkName()
+            const clientVersionProm = web3Api.web3.currentProvider.send('web3_clientVersion', [])
```

Comment:
I think there was already a web3 method for this.

Anyways. How do you handle the `null provider` thing we are discussing in the other thread. 
As I wrote in the other thead. If the wallet is not connected. I think it's better to have a null `web3Api` and handle that case.

If the wallet re-connects, this component function will be called again with a non-null value

Reply:
No universal method for that. Won't work with some wallets, and this method will

Reply:
But one of my comments is, I saw in other parts of the PR that `web3Api.web3.currentProvider` can be `null` in some cases. 

This will crash if it's the case, right?

Anyways, I also think is nor right to do the `new Web3(null)`, we should handle the case differently too (there's other conversation for this topic)

Also, I would argue that the `web3Api.web3.currentProvider.send(...)` is a bit coupled, and since you already have an interface for interacting `web3Api` it would be simpler and more readable to do sth like `web3Api.sendRpc(...)`